### PR TITLE
Refactor: Apply copier template

### PR DESCRIPTION
**Pull request by adrien-berchet**
Major refactoring to comply with OSS requirements.

### CODE REVIEW

**Review by arnaudon — APPROVED** [review]

File: tmd/Topology/analysis.py [hunk 1]
```diff
@@ -201,8 +189,12 @@ def barcode_bin_centers(ph, num_bins=100, min_bin=None, max_bin=None):
...
 
-    data = [bin_centers[i] for p in ph_2D for i in range(num_bins - 1)
-            if np.max(p) > bins[i + 1] and np.min(p) < bins[i]]
+    data = [
+        bin_centers[i]
+        for p in ph_2D
+        for i in range(num_bins - 1)
+        if np.max(p) > bins[i + 1] and np.min(p) < bins[i]
+    ]
     # data = []
```
Comment (arnaudon) [inline tmd/Topology/analysis.py:198]: is this useful?

**Comment by arnaudon** [comment]
I didn't see anything major, maybe after you merge, ask alexander to have a look for OS requirements? Probably the doc will need some fine tuning as well in later PRs.

File: tmd/Topology/analysis.py (hunk 1 above)
Comment (arnaudon) [reply tmd/Topology/analysis.py:198]: I mean the commented code starting from this line

Comment (adrien-berchet) [reply tmd/Topology/analysis.py:214]: Obviously not :smile: 
But I was not sure why it was there so I kept it. I will add a TODO to mark it. Or maybe @lidakanari thinks we can just remove it?

Comment (lidakanari) [reply tmd/Topology/analysis.py:215]: the 206? No, you can remove it :)

**Review by lidakanari — APPROVED** [review]
I can't say I managed to read everything but I tried the code and nothing is breaking and from a quick read-through it LGTM

File: tmd/Topology/analysis.py (hunk 1 above)
Comment (adrien-berchet) [reply tmd/Topology/analysis.py]: Ok, thanks!

**Comment by adrien-berchet** [comment]
> I can't say I managed to read everything but I tried the code and nothing is breaking and from a quick read-through it LGTM

Ok cool, thanks for the review.

Let's merge it and then I will ask @alex4200 to check for OSS requirements.

**Pull request merged**